pkg/controller/deployment/cert_manager_webhook_deployment.go

```diff
@@ -0,0 +1,31 @@
+package deployment
+
+import (
+	"github.com/openshift/cert-manager-operator/pkg/operator/kubeclient"
+	configv1 "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/status"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	"k8s.io/client-go/informers"
+)
+
+var (
+	certManagerWebhookDeploymentFile = "cert-manager-deployment/cert-manager-webhook-deployment.yaml"
+	certManagerWebhookAssetFiles     = []string{
+		// FIXME: Add Mutating Webhook Configuration into github.com/openshift/library-go/pkg/operator/resource/resourceapply/unstructured.go
+		// Then, this these guys back on.
+		//"cert-manager-deployment/cert-manager-webhook-mutatingwebhookconfiguration.yaml",
+		//"cert-manager-deployment/cert-manager-webhook-validatingwebhookconfiguration.yaml",
+		"cert-manager-deployment/cert-manager-webhook-sa.yaml",
+		"cert-manager-deployment/cert-manager-webhook-svc.yaml",
+		"cert-manager-deployment/cert-manager-webhook:dynamic-serving-rb.yaml",
+		"cert-manager-deployment/cert-manager-webhook:dynamic-serving-role.yaml",
+		"cert-manager-deployment/cert-manager-webhook:subjectaccessreviews-cr.yaml",
+		"cert-manager-deployment/cert-manager-webhook:subjectaccessreviews-crb.yaml",
+	}
+)
+
+func NewCertManagerWebhookDeployment(operatorNamespace, targetNamespace, targetVersion string, operatorClient v1helpers.OperatorClient, kubeClientContainer kubeclient.KubeClientContainer, kubeInformersForTargetNamespace informers.SharedInformerFactory, openshiftClusterConfigClient configv1.ClusterOperatorInterface, eventsRecorder events.Recorder, versionRecorder status.VersionGetter) factory.Controller {
```

Comment:
I don't know what this is. It's missing godoc.

Reply:
I refactored this piece of code and hope it will be self-explanatory now.